src/pip/_internal/utils/misc.py

```diff
@@ -612,7 +612,7 @@ class cached_property(object):
     """
 
     def __init__(self, func):
-        self.__doc__ = getattr(func, '__doc__')
+        self.__doc__ = getattr(func, '__doc__', None)
```

Comment:
Same here:

```suggestion
        self.__doc__ = func.__doc__
```

Reply:
On other thought, the `cached_property` class where this change was done is being removed in https://github.com/pypa/pip/pull/8403/ , so I might not need to make that change at all.

Should I wait for that PR to be merged and then this change won't be needed (I see that you have approved it, so maybe it can be merged)?

Reply:
Now that this has been rebased to master with the changes in #8403 , this comment no longer applies.